app/scripts/components/common/page-header/use-mobile-menu-fix.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+
+/**
+ * Would be better to have this fixed on the react-uswds side!
+ *
+ * Subscribe to https://github.com/trussworks/react-uswds/issues/2586
+ * for updates
+ * */
+
+const USWDS_DESKTOP_BREAKPOINT = 1024;
+// no magic numbers! Use value from theme?
+
+function useMobileMenuFix(expanded, setExpanded) {
+  const handleClickOutside = (event: MouseEvent) => {
+    // When the mobile nav is active, clicking outside of the nav should close it
+    if (
+      event.target instanceof Element &&
+      event.target.closest('.usa-overlay')
+    ) {
+      setExpanded(false);
+    }
+  };
+
+  const handleResize = () => {
+    if (window.innerWidth > USWDS_DESKTOP_BREAKPOINT && expanded) {
+      setExpanded(false);
```

Comment:
To also hide the overlay on resize, maybe we could also set the `showMobileOverlay` that [exists on the React USWDS component](https://trussworks.github.io/react-uswds/?path=/story/components-header--extended-header) to `false`? Not sure if that prop works as intended though, but worth a try.

Reply:
Or we could just remove the `is-visible` class from the `usa-overlay` when the screen is resized?

<img width="379" alt="Screenshot 2024-11-26 at 12 13 17" src="https://github.com/user-attachments/assets/77bc537c-9baa-4a15-8521-2737e5dbcab9">

Reply:
The expanded state is being passed to `showMobileOverlay={expanded}` in the USWDSHeader props, thats how the overlay is hidden :) Does it not work for you?